packages/expo-sqlite/src/paramUtils.ts

```diff
@@ -94,3 +94,39 @@ export function composeRows<T>(
   }
   return results;
 }
+
+/**
+ * Normalize the index for the Storage.getKeyByIndexSync and Storage.getKeyByIndexAsync methods.
+ * @returns The normalized index or `null` if the index is out of bounds.
+ * @hidden
+ */
+export function normalizeStorageIndex(index: any): number | null {
+  // Convert the index to a number
+  let value: number;
+  if (typeof index === 'boolean') {
+    value = index ? 1 : 0;
+  } else if (typeof index === 'string') {
+    value = Number(index);
+  } else if (typeof index === 'bigint') {
+    value = Number(index);
+  } else if (typeof index.valueOf === 'function') {
+    value = index.valueOf();
+  } else {
+    value = index;
+  }
```

Comment:
I think this could just be:
```suggestion
  const value = Number(index);
```

`Number(true/false) => 0/1`
`Number({ valueOf: () => 123 }) => 123`
`Number({ toString:() => '123' }) => 123`

I tested them and they match the behavior of `localStorage.key(x)` in Chrome.